bpf/tc_http_tp.c

```diff
@@ -0,0 +1,588 @@
+#include <vmlinux.h>
+
+#include <bpf_endian.h>
+#include <bpf_helpers.h>
+#include <bpf_tracing.h>
+
+#include "http_types.h"
+#include "tcp_info.h"
+#include "tracing.h"
+
+char LICENSE[] SEC("license") = "Dual BSD/GPL";
+
+enum { TC_ACT_OK = 0 };
+enum { MAX_IP_PACKET_SIZE = 0x7fff };
+enum { MAX_INLINE_LEN = 0x7ff };
+
+struct seq_offset_map {
+    __uint(type, BPF_MAP_TYPE_HASH);
+    __type(key, __u32);
```

Comment:
This is currently only indexing on the ephemeral port - I reckon we require something a bit smarter here.